src/main.h

```diff
@@ -332,6 +333,10 @@ bool ContextualCheckInputs(const CTransaction& tx, CValidationState &state, cons
                            unsigned int flags, bool cacheStore, const Consensus::Params& consensusParams,
                            std::vector<CScriptCheck> *pvChecks = NULL);
 
+/** Check a transaction contextually against a set of consensus rules */
+bool ContextualCheckTransaction(const CTransaction& tx, CValidationState &state,
+                                int nHeight = 0, int dosLevel = 0);
```

Comment:
Is there anywhere we don't want to pass in a height? DoS level I can understand not passing in (in tests).

Reply:
I'm going to remove both defaults.  The caller must provide the context which also includes dos level.